drivers/soundwire/intel.c

```diff
@@ -401,6 +401,41 @@ static void intel_shim_wake(struct sdw_intel *sdw, bool wake_enable)
 	}
 }
 
+void sdw_intel_jack_detect(struct sdw_intel_ctx *ctx)
+{
+	struct sdw_intel_link_res *link;
+	struct sdw_master_device *md;
+	struct sdw_slave *slave;
+	struct sdw_bus *bus;
+
+	list_for_each_entry(link, &ctx->link_list, list) {
+		struct sdw_intel *sdw;
+		u16 wake_sts;
+
+		md = link->md;
+		bus = &link->cdns->bus;
+		sdw = cdns_to_intel(link->cdns);
+		wake_sts = intel_readw(sdw->link_res->shim, SDW_SHIM_WAKESTS);
+
+		if (!(wake_sts & BIT(sdw->instance)))
+			continue;
+
+		/*
+		 * wake up master and slave so that slave can notify master
+		 * the jack event and let codec driver check jack status
+		 */
+		list_for_each_entry(slave, &bus->slaves, node) {
+			if (slave->prop.wake_capable) {
+				if (slave->status != SDW_SLAVE_ATTACHED)
```

Comment:
could be in alert already?

Reply:
our ACPI table reports 700, 701 & 711 on CML, so these slaves will be added but only 711 is attached.

Reply:
compared to normal clock stop mode, master is powered off in our clock stop implementation, so master can't response to slave. Instead, glue hw will get the wake-up event and report a PME event to PCI to wake up pci driver (sof-audio).   sof-audio driver will wake up master and slave to check event

Reply:
it doesn't matter what the master does.
The point is that the slave can report in ALERT from the first PING frame, this is a different state machine. The master cannot control what the slave does.